extensions/npm/main.ts

```diff
@@ -1,32 +1,85 @@
-import { PhylumApi } from "phylum";
-
-// Parse CLI args.
-const args = Deno.args;
-if (args.length != 2 || args[0] !== "install") {
-    console.error("Usage: phylum npm install <PKG>");
-} else {
-    await install(args[1]);
+import { red, green, yellow } from 'https://deno.land/std@0.150.0/fmt/colors.ts';
+import { PhylumApi } from 'phylum';
+
+class FileBackup {
+  readonly fileName: string
+  readonly fileContent: string | null
+
+  constructor(fileName: string) {
+    this.fileName = fileName
+    this.fileContent = null
+  }
+
+  async backup() {
+    try {
+      this.fileContent = await Deno.readTextFile(this.fileName)
+    } catch (e) {}
+  }
+
+  async restoreOrDelete() {
+    try {
+      if (this.fileContent != null) {
+        await Deno.writeTextFile(this.fileName, this.fileContent)
+      } else {
+        await Deno.remove(this.fileName)
+      }
+    } catch (e) {}
+  }
 }
 
-// Analyze and install package.
-async function install(pkg: string) {
-    console.log("Updating package lock…");
-    await Deno.run({ cmd: ["npm", "i", "--package-lock-only", pkg] }).status();
-    console.log("Package lock updated.\n");
+// Analyze new dependencies with phylum before install/update.
+if (Deno.args.length >= 1
+    && (
+        'install'.startsWith(Deno.args[0])
+        || 'isntall'.startsWith(Deno.args[0])
+        || 'update'.startsWith(Deno.args[0])
+        || 'udpate'.startsWith(Deno.args[0])
+    )) {
+    await checkDryRun(Deno.args[0], Deno.args.slice(1));
+}
 
-    console.log("Analyzing packages…");
-    const jobId = await PhylumApi.analyze("./package-lock.json");
-    const jobStatus = await PhylumApi.getJobStatus(jobId);
+// Run the command with side effects.
+console.log('Applying changes…');
+let status = await Deno.run({ cmd: ['npm', ...Deno.args] }).status();
+Deno.exit(status.code);
```

Comment:
I know JS has hoisting, but I'd move this at the bottom of the file. Having this part here between `checkDryRun` and `FileBackup` feels a bit inconsistent semantically.